core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -62,7 +62,8 @@ abstract class AbstractFetcherThread(name: String,
   type EpochData = OffsetsForLeaderEpochRequest.PartitionData
 
   private val partitionStates = new PartitionStates[PartitionFetchState]
-  private val partitionMapLock = new ReentrantLock
+  // Visible for testing see ReplicaManagerTest#testFencedErrorCausedByBecomeLeader
+  private[server] val partitionMapLock = new ReentrantLock
```

Comment:
At a high level, I am concerned with the stability of this test as we make changes to the code in the future. It seems like we are exposing internal concurrency requirements to the tests.

Reply:
Got it. Let me revert this change.